keras_cv/models/generative/stable_diffusion/stable_diffusion.py

```diff
@@ -89,10 +101,16 @@ def __init__(self, img_height=512, img_width=512, jit_compile=False):
         self.text_encoder = TextEncoder(MAX_PROMPT_LENGTH)
         self.diffusion_model = DiffusionModel(img_height, img_width, MAX_PROMPT_LENGTH)
         self.decoder = Decoder(img_height, img_width)
+
+        self.input_img_height = input_img_height
+        self.input_img_width = input_img_width
+        self._image_encoder = None
+
         if jit_compile:
             self.text_encoder.compile(jit_compile=True)
             self.diffusion_model.compile(jit_compile=True)
             self.decoder.compile(jit_compile=True)
+            self.image_encoder.compile(jit_compile=True)
```

Comment:
If I understand correctly, this causes us to instantiate an image encoder model whenever jit_compile is being used (regardless of whether it's actually going to be used). I think this is probably not the behavior we want.

Reply:
Oh yes, that is correct.  Good catch!